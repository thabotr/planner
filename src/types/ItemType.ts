class ItemType {
    length: number;
    mES: number;
    pES: number;
    id: string;
    constructor(length: number, mES: number, pES: number, id: string) {
        this.length = length;
        this.id = id;
        this.mES = mES;
        this.pES = pES;
    }
};

export default ItemType;