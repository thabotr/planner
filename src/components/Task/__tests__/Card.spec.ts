import GenericCardVue from "@/components/Task/GenericCard.vue";
import ItemType from "@/types/ItemType";
import { describe, expect, it } from "vitest";
import { CustomVueWrapper } from "../../__tests__/utils";

describe('Card', () => {
    const wrapper = new CustomVueWrapper(GenericCardVue, {
        props: {
            item: new ItemType(1_000_000, 19, 70, "id"),
        },
        slots: {
            default: 'Default',
            actions: 'Actions go here',
            controls: 'Card effort controls go here',
        }
    });
    it('renders correctly', () => {
        expect(wrapper.wrapper.html()).toMatchSnapshot();
    });
})