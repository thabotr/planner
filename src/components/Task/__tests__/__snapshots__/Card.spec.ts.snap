// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > renders correctly 1`] = `
"<div data-v-f39f1573=\\"\\">
  <div data-v-f39f1573=\\"\\" class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\" id=\\"bg\\">
    <!---->
    <div class=\\"v-card__loader\\">
      <div class=\\"v-progress-linear v-theme--light\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
        <!---->
        <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
        <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
          <div class=\\"v-progress-linear__indeterminate\\">
            <div class=\\"v-progress-linear__indeterminate long\\"></div>
            <div class=\\"v-progress-linear__indeterminate short\\"></div>
          </div>
        </transition-stub>
        <!---->
      </div>
    </div>
    <!---->
    <!---->
    <div data-v-f39f1573=\\"\\">
      <div data-v-f39f1573=\\"\\" class=\\"v-card v-theme--light v-card--density-default v-card--variant-elevated\\" id=\\"fg\\">
        <!---->
        <div class=\\"v-card__loader\\">
          <div class=\\"v-progress-linear v-theme--light\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px; left: 50%; transform: translateX(-50%);\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
            <!---->
            <div class=\\"v-progress-linear__background\\" style=\\"width: 100%;\\"></div>
            <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
              <div class=\\"v-progress-linear__indeterminate\\">
                <div class=\\"v-progress-linear__indeterminate long\\"></div>
                <div class=\\"v-progress-linear__indeterminate short\\"></div>
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <!---->Default<div data-v-f39f1573=\\"\\" class=\\"flex-horizontal\\">
          <div data-v-19720537=\\"\\" data-v-f39f1573=\\"\\" class=\\"flex-vertical justify-space-between\\">
            <div data-v-19720537=\\"\\" class=\\"flex-horizontal\\"><i data-v-19720537=\\"\\" class=\\"mdi-brain mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\" id=\\"i-mental\\"></i>1 mental EP </div>
            <div data-v-19720537=\\"\\" class=\\"flex-horizontal\\"><i data-v-19720537=\\"\\" class=\\"mdi-account-hard-hat mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\" id=\\"i-physical\\"></i>21 physical EP </div>
            <div data-v-19720537=\\"\\" class=\\"flex-horizontal\\"><i data-v-19720537=\\"\\" class=\\"mdi-timer mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\" id=\\"i-time\\"></i>16m</div>
          </div>Card effort controls go here
        </div>
        <!---->
        <!----><span class=\\"v-card__underlay\\"></span>
      </div>
    </div>
    <div data-v-f39f1573=\\"\\" id=\\"actions\\" class=\\"padded-s\\"><span data-v-f39f1573=\\"\\" class=\\"v-chip v-theme--light v-chip--density-compact v-chip--size-default v-chip--variant-tonal\\" draggable=\\"false\\"><!----><span class=\\"v-chip__underlay\\"></span>
      <!---->
      <!---->id
      <!---->
      <!----></span>Actions go here
    </div>
    <!---->
    <!----><span class=\\"v-card__underlay\\"></span>
  </div>
</div>"
`;
